server_common/snmpWalker.py

```diff
@@ -0,0 +1,64 @@
+from pysnmp.hlapi import *
+import sys
+from pysnmp.smi import builder, view, compiler, rfc1902
+
+from server_common.utilities import print_and_log, SEVERITY
+
+# Assemble MIB browser
+mibBuilder = builder.MibBuilder()
+mibViewController = view.MibViewController(mibBuilder)
+compiler.addMibCompiler(mibBuilder, sources=['file:///usr/share/snmp/mibs',
+                                             'https://mibs.pysnmp.com/asn1/@mib@'])
+
```

Comment:
Do downloaded compiled MIBS get cached and re-used next time? Do we need to pass the `destination=` keyword to do this? If they do not get re-used then maybe we need to download them ourselves and refer to then via `sources`, i didn't want to see the ibex process hang if e.g.   `mibs.pysnmp.com` was offline

Reply:
Yes, probably using destination means the MIBs are downloaded and can be reused later. Let me try and see the results